src/test/java/org/mskcc/cbio/oncokb/web/rest/UsageAnalysisControllerIT.java

```diff
@@ -0,0 +1,745 @@
+      )
+      // strict checking enforces the same order on arrays
+      // we need to use strict checks to make sure we are not missing
+      // any data in our asserts
+      // if you have issues with arrays make sure you have the correct order
+      .andExpect(content().json(expected, true));
+  }
+
+  @Test
+  public void shouldGetUsageSummaryUsers() throws Exception {
+    String url = "/api/usage/summary/users";
+    String expected = gson.toJson(data.files.get(url));
+    restMockMvc
+      .perform(get(url))
+      .andExpect(status().isOk())
+      .andExpect(
+        content().contentTypeCompatibleWith(MediaType.APPLICATION_JSON)
+      )
+      .andExpect(content().json(expected, true));
+  }
+
+  @Test
+  public void shouldGetUsageSummaryUsersByCompanyId() throws Exception {
+    String url =
+      "/api/usage/summary/users?companyId=" +
+      data.userDtos.get(0).getCompany().getId();
+    String expected = gson.toJson(data.files.get(url));
+    restMockMvc
+      .perform(get(url))
+      .andExpect(status().isOk())
+      .andExpect(
+        content().contentTypeCompatibleWith(MediaType.APPLICATION_JSON)
+      )
+      .andExpect(content().json(expected, true));
+  }
+
+  @Test
+  public void shouldGetUsageUsersById() throws Exception {
+    String url = "/api/usage/users/" + data.users.get(0).getId();
+    String expected = gson.toJson(data.files.get(url));
+    restMockMvc
+      .perform(get(url))
+      .andExpect(status().isOk())
+      .andExpect(
+        content().contentTypeCompatibleWith(MediaType.APPLICATION_JSON)
+      )
+      .andExpect(content().json(expected, true));
+  }
+
+  private static class MockS3Data {
+
+    private static class Tuple<X, Y> {
+      public final X first;
+      public final Y second;
+
+      public Tuple(X first, Y second) {
+        this.first = first;
+        this.second = second;
+      }
+    }
+
+    private final String[] sampleUrls = {
+      "/api/v1/annotate/mutations/byProteinChange",
+      "/api/private/utils/numbers/main/",
+      "/api/v1/annotate/structuralVariants",
+      "/api/private/utils/numbers/levels/",
+      "/api/v1/annotate/copyNumberAlterations",
+      "/api/private/search/typeahead",
+    };
+    private final String[] fakeCompanies = { "FakeCo", "DemoLLC" };
+    private final Random random;
+
+    private final JsonObject files = new JsonObject();
+
+    private final List<User> users = new ArrayList<>();
+    private final List<UserDTO> userDtos = new ArrayList<>();
+
+    public MockS3Data(int seed) throws Exception {
+      this(seed, new String[] { "doej", "smithj" });
+    }
+
+    public MockS3Data(int seed, String[] userNames) throws Exception {
+      random = new Random(seed);
+      addUsers(userNames);
+    }
+
+    private String getNextSampleUrl() {
+      return sampleUrls[Math.abs(random.nextInt() % sampleUrls.length)];
+    }
+
+    private Tuple<String, JsonObject> fetchJsonObjectFromFilesObject(
```

Comment:
I'm a bit confused about this method. This is to fetch the last key json but backfill all keys if they do not exist in obj?  Maybe add some comments here?

Reply:
That is correct, but I'll add some comments.